internal/controller/cluster/reconciler.go

```diff
@@ -1023,3 +1034,48 @@ func (r *SingleClusterReconciler) AddAPIVersionLabel(ctx context.Context) error
 
 	return r.Client.Update(ctx, aeroCluster, common.UpdateOption)
 }
+
+func (r *SingleClusterReconciler) IsReclusterNeeded() bool {
+	// Return false if dynamic configuration updates are disabled
+	if !asdbv1.GetBool(r.aeroCluster.Spec.EnableDynamicConfigUpdate) {
+		return false
+	}
+
+	for specIdx := range r.aeroCluster.Spec.RackConfig.Racks {
+		for statusIdx := range r.aeroCluster.Status.RackConfig.Racks {
+			if r.aeroCluster.Spec.RackConfig.Racks[specIdx].ID == r.aeroCluster.Status.RackConfig.Racks[statusIdx].ID &&
```

Comment:
Looks like a complicated condition, comments would help understand the condition.